core/src/test/java/org/opensearch/sql/planner/physical/TakeOrderedOperatorTest.java

```diff
@@ -0,0 +1,603 @@
+      new HashMap<>() {
+        {
+          put("size", 399);
+          put("response", null);
+        }
+      };
+
+  @Test
+  public void sort_one_field_asc() {
+    List<ExprValue> inputList =
+        Arrays.asList(
+            tupleValue(ImmutableMap.of("size", 499, "response", 404)),
+            tupleValue(ImmutableMap.of("size", 320, "response", 200)),
+            tupleValue(ImmutableMap.of("size", 399, "response", 503)));
+
+    wrapper.setIterator(inputList.iterator());
+    assertThat(
+        execute(
+            takeOrdered(
+                inputPlan, 3, 0, Pair.of(SortOption.DEFAULT_ASC, ref("response", INTEGER)))),
+        contains(
+            tupleValue(ImmutableMap.of("size", 320, "response", 200)),
+            tupleValue(ImmutableMap.of("size", 499, "response", 404)),
+            tupleValue(ImmutableMap.of("size", 399, "response", 503))));
+
+    wrapper.setIterator(inputList.iterator());
+    assertThat(
+        execute(
+            takeOrdered(
+                inputPlan, 2, 0, Pair.of(SortOption.DEFAULT_ASC, ref("response", INTEGER)))),
+        contains(
+            tupleValue(ImmutableMap.of("size", 320, "response", 200)),
+            tupleValue(ImmutableMap.of("size", 499, "response", 404))));
+
+    wrapper.setIterator(inputList.iterator());
+    assertThat(
+        execute(
+            takeOrdered(
+                inputPlan, 2, 1, Pair.of(SortOption.DEFAULT_ASC, ref("response", INTEGER)))),
+        contains(
+            tupleValue(ImmutableMap.of("size", 499, "response", 404)),
+            tupleValue(ImmutableMap.of("size", 399, "response", 503))));
+  }
+
+  @Test
+  public void sort_one_field_with_duplication() {
+    List<ExprValue> inputList =
+        Arrays.asList(
+            tupleValue(ImmutableMap.of("size", 499, "response", 404)),
+            tupleValue(ImmutableMap.of("size", 320, "response", 404)),
+            tupleValue(ImmutableMap.of("size", 399, "response", 503)));
+
+    wrapper.setIterator(inputList.iterator());
+    assertThat(
+        execute(
+            takeOrdered(
+                inputPlan, 3, 0, Pair.of(SortOption.DEFAULT_ASC, ref("response", INTEGER)))),
+        contains(
+            tupleValue(ImmutableMap.of("size", 499, "response", 404)),
+            tupleValue(ImmutableMap.of("size", 320, "response", 404)),
+            tupleValue(ImmutableMap.of("size", 399, "response", 503))));
+
+    wrapper.setIterator(inputList.iterator());
+    assertThat(
+        execute(
+            takeOrdered(
+                inputPlan, 2, 0, Pair.of(SortOption.DEFAULT_ASC, ref("response", INTEGER)))),
+        contains(
+            tupleValue(ImmutableMap.of("size", 499, "response", 404)),
+            tupleValue(ImmutableMap.of("size", 320, "response", 404))));
+
+    wrapper.setIterator(inputList.iterator());
+    assertThat(
+        execute(
+            takeOrdered(
+                inputPlan, 2, 1, Pair.of(SortOption.DEFAULT_ASC, ref("response", INTEGER)))),
```

Comment:
Could you add a test which its limit equals its offset?

Reply:
Oh, it comes to me that we may not do that test since Limit(offset, limit) in OpenSearch has different semantics with other engine. For OpenSearch, limit stands for the size of `HEAD` and offset stands for the from of `HEAD`.
So it will always return `limit` rows no matter which value `offset` is.
e.g. 
```
POST _plugins/_ppl
{
  "query": """
    source=opensearch_dashboards_sample_data_flights| fields FlightTimeMin | head 2 from 2
  """
}

# response
{
  "schema": [
    {
      "name": "FlightTimeMin",
      "type": "float"
    }
  ],
  "datarows": [
    [
      0
    ],
    [
      222.74905
    ]
  ],
  "total": 2,
  "size": 2
}

```

Reply:
> So it will always return limit rows no matter which value offset is.

Will `skip(offset)` in `open()` method of TakeOrderedOperator` change the current behavior?
It comes to me better to add a test to compare the results between take-ordered and limit+head.

Reply:
Sorry, it's not exact right to say `it will always return limit rows no matter which value offset is` actually. It's right only when the size of input rows is more than `offset + limit`. 

I think `skip(offset)` is semantic equal to what LimitOperator do in its `open()`. I'll add test to verify this.